server/ServerFSM.cpp

```diff
@@ -468,6 +482,13 @@ MPLobby::MPLobby(my_context c) :
                     player_setup_data.m_starting_species_name = sm.SequentialPlayableSpeciesName(player_id);
 
                 m_lobby_data->m_players.push_back(std::make_pair(player_id, player_setup_data));
+
+                player_connection->SetAuthRoles({
+                                Networking::ROLE_MODERATOR,
```

Comment:
most players aren't moderators or observers

Reply:
Currently they can set yourself to Observer or Moderator so I just leave it.

Reply:
So the `SetAuthRoles` function is setting the allowed roles, not the current roles? If so, rename it.

Reply:
They are current roles. Player can set itself to Moderator, Observer or Player in any time so he has ROLE_MODERATOR, ROLE_OBSERVER and ROLE_PLAYER.

Reply:
If that's how it works, then the term "role" is confusing.